packages/ui/src/models/visualization/flows/support/camel-steps.service.ts

```diff
@@ -17,9 +17,10 @@ export class CamelStepsService {
     componentLookup: ICamelElementLookupResult,
     entityDefinition: unknown,
   ): IVisualizationNode {
+    const nodeIconType = componentLookup.componentName ? NodeIconType.Component : NodeIconType.EIP;
     const data: CamelRouteVisualEntityData = {
       path,
-      icon: NodeIconResolver.getIcon(CamelComponentSchemaService.getIconName(componentLookup)),
+      icon: NodeIconResolver.getIcon(CamelComponentSchemaService.getIconName(componentLookup), nodeIconType),

```

Comment:
This needs further evaluation, as we need to confirm that it works well with Kamelets as well.